### PR TITLE
create QgsMapLayer::LayerFlags for Searchable, Identifiable and Removable

### PR DESCRIPTION
**Pull request by 3nids**
this brings back the information previously saved in the project node back to the layer node
this will allow to embed this information in layer styles

@wonder-sk you added the non removable list to the project very recently. After discussing with @nyalldawson and @m-kuhn, I thought it should lie in the map layer rather than in the project. Are you ok with this?

### CODE REVIEW

**Review by wonder-sk — APPROVED** [review]
I like this unification of flags that were spread around!

File: src/core/qgsmaplayer.h [hunk 1]
```diff
@@ -146,6 +156,22 @@ class CORE_EXPORT QgsMapLayer : public QObject
      */
     QgsMapLayer::LayerType type() const;
 
+    /**
+     * Returns the flags for this layer.
+     * \note Flags are options specified by the user and might not reflect the actual state.
+     * For instance, even if the ReadOnly flag is not set, a raster layer will always be ready-only.
```
[the remark this answers is not in the document]
Comment (3nids) [reply src/core/qgsmaplayer.h]: had to remove this because the vector layer setter needs to do its own receipe

**Comment by 3nids** [comment]
@wonder-sk thanks for the review!

**Comment by 3nids** [comment]
now good to go, only WMS test failing

**Review by nyalldawson — APPROVED** [review]
API looks fantastic to me! I've only given the code a general read over and it looks solid to me, I'll trust you've got the details right ;)

**Comment by 3nids** [comment]
Thanks for the review

**Pull request merged**